modules/models/record.php

```diff
@@ -112,6 +112,44 @@ public static function find_by($attrs)
         }
     }
 
+    public function delete($pk="id") // Deletes by assuming $id is always the pk. Not sure if robust enough.
+    {
+        // Check to see if we are deleting a record that has not been saved yet.
+        if ($this->is_new_record) {
+            throw new ErrorException("Cannot delete an object that is not yet in a table. Please set the object to null instead.");
+        }
+
+        $attrs = $this->getAttrs();
+
+        // Ensure that the pk exists
+        if (!in_array($pk, $attrs)) {
+            throw new ErrorException("Record trying to delete using a primary key that does not exist: ".$pk);
+        }
+
+        $result = get_called_class()::where(array($pk=>$this->$pk));
+        // Make sure that only one record with the supplied key exists in the table
+        if (count($result) > 1) {
+            throw new ErrorException("Primary key specified is not valid as it returns more than one record. PK: ".$pk);
+        }
+
+        $condition = sprintf("%s = %s", $pk, $this->$pk);
```

Comment:
If the pk is not numeric then this fails I think

Reply:
good catch, ill fix that